scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -171,6 +171,16 @@ class ScioContext private[scio] (val options: PipelineOptions,
     o.setScioVersion(scioVersion)
   }
 
+  {
+    /** Check if running within [[scala.App]] see https://github.com/spotify/scio/issues/449 */
+    if (Thread.currentThread()
+      .getStackTrace.toList.map(_.getClassName.split('$').head)
+      .exists(_.equals(classOf[App].getName))) {
+      logger.warn(
+        "Applications declared within scala.App might not work properly, please use main method!")
```

Comment:
grammar nazi: "defined" instead of "declared" (which suggests no impl). ". Please"